### PR TITLE
movie lens with cofire

### PR DESCRIPTION
**Pull request by db7**

### CODE REVIEW

File: refeeder.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package cofire
+
+import (
+	fmt "fmt"
+	"time"
+
+	"github.com/facebookgo/clock"
```
Comment (heltonmarx) [inline refeeder.go]: We actually really need this `facebookgo/clock` ? I think that we can use the current STL `time` and adapt the unit tests for it.

**Comment by sweigert** [comment]
I think a small architecture figure would help. It is hard to understand, for example what the refeeder is doing.

File: refeeder_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	// a message arrives at start timestamp
+	clk.Add(1 * time.Hour)
+	start := clk.Now()
+
+	// the message was created at start - 10 seconds
+	ctx := new(mockContext)
+	ctx.ts = start.Add(-10 * time.Second)
+	ctx.emitCheck = func(s goka.Stream, k string, m interface{}) {
+		ensure.DeepEqual(t, s, goka.Stream("topic"))
+		ensure.DeepEqual(t, k, "key")
```
Comment (heltonmarx) [inline refeeder_test.go]: STL tests here ! what do you think ?

Comment (db7) [reply refeeder_test.go]: yes yes yes... I just forgot it again :-)

File: refeeder.go (hunk 1 above)
Comment (db7) [reply refeeder.go]: mhhh, well the clock is pretty useful to test the time behavior. How would you adapt the test?

File: refeeder_test.go (hunk 2 above)
Comment (db7) [reply refeeder_test.go]: done

File: refeeder.go (hunk 1 above)
Comment (db7) [reply refeeder.go]: done

**Comment by db7** [comment]
@sweigert you are right. I will try to come up with something in another PR.

File: refeeder.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package cofire
+
+import (
+	fmt "fmt"
```
Comment (heltonmarx) [inline refeeder.go]: I think this is not necessary...`fmt "fmt"`, a simple `fmt` is good.

Comment (db7) [reply refeeder.go]: done.

also using `time.After` now.

**Review by heltonmarx — APPROVED** [review]

**Pull request merged**